src/main/java/com/android/volley/CacheDispatcher.java

```diff
@@ -64,6 +71,7 @@ public CacheDispatcher(
         mNetworkQueue = networkQueue;
         mCache = cache;
         mDelivery = delivery;
+	mWaitingRequestHandler = new WaitingRequestHandler();
```

Comment:
Declare static nested so the interactions with CacheDispatcher are explicit: `WaitingRequestHandler(this)`